sdb/commands/zfs/histograms.py

```diff
@@ -0,0 +1,149 @@
+#
+# Copyright 2020 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+from typing import Iterable
+
+import drgn
+import sdb
+from sdb.commands.internal import fmt
+
+
+class ZFSHistogram(sdb.Command):
+    """
+    Print ZFS Histogram and print its median segment size.
+
+    NOTE
+        The median is just an approximation as we can't tell the
+        exact size of each bucket within a histogram bucket.
+
+    EXAMPLES
+        Dump the histogram of the normal metaslab class of the rpool:
+
+            sdb> spa rpool | member spa_normal_class.mc_histogram | zhist
+            idx: counts
+            ---- ------
+              9:   4264 ************************
+             10:   3168 ******************
+             11:   3545 ********************
+             12:   3344 *******************
+             13:   3724 *********************
+             14:   4184 ***********************
+             15:   7221 ****************************************
+             16:   3942 **********************
+             17:   2408 *************
+             18:   1558 *********
+             19:    979 *****
+             20:    538 ***
+             21:    390 **
+             22:    228 *
+             23:     97 *
+             24:     46
+             25:     22
+             26:      8
+             27:      9
+             28:     97 *
+            Histogram Median: 343.4MB
+    """
+
+    names = ["zfs_histogram", "zhist"]
+
+    @classmethod
+    def _init_parser(cls, name: str) -> argparse.ArgumentParser:
+        parser = super()._init_parser(name)
+        parser.add_argument("offset", nargs="?", default=0, type=int)
+        return parser
+
+    @staticmethod
+    def histogram_median(hist: drgn.Object, offset: int = 0) -> int:
+        """
+        Returns the approximated median of a ZFS histogram.
+        """
+        canonical_type = sdb.type_canonicalize(hist.type_)
+        assert canonical_type.kind == drgn.TypeKind.ARRAY
+        assert sdb.type_canonicalize(
+            canonical_type.type).kind == drgn.TypeKind.INT
+        total_space, max_bucket, min_bucket = 0, 0, (len(hist) - 1)
+
+        for (bucket, value) in enumerate(hist):
+            count = int(value.value_())
```

Comment:
I think that we don't need `.value_()` here (and elsewhere).  From the drgn docs:

> The standard int(), float(), and bool() functions convert an object to that Python type. Conversion to bool uses the programming language’s notion of “truthiness”. Additionally, certain Python functions will automatically coerce an object to the appropriate Python type (e.g., hex(), round(), and list subscripting).